app/controllers/hub/documents_controller.rb

```diff
@@ -20,35 +20,36 @@ def show
       redirect_to transient_storage_url(@document.upload.blob)
     end
 
+    def new; end
+
     def edit; end
 
     def create
       document_params[:upload].each do |file_upload|
         Document.create!(
           client: @client,
           intake: @client.intake,
-          document_type: DocumentTypes::Other.key,
-          upload: file_upload
+          document_type: document_params[:document_type],
+          upload: file_upload,
+          display_name: document_params[:display_name],
+          tax_return_id: document_params[:tax_return_id]
         )
       end
       redirect_to(hub_client_documents_path(client_id: @client))
     end
 
     def update
-      @form = Hub::DocumentForm.new(@document, document_params)
-      if @form.valid?
-        @form.save
+      if @document.update(document_params)
```

Comment:
This will pass the upload param (which is a multi select) when updating the document. Does that cause an issue? Does it clear out the existing attachment and add another? Asking because in the create method, we're having to loop through the upload value to create individual documents and here we're passing the array in as a value to upload. Seems like it could cause unexpected behavior, but maybe not!

Reply:
You're right, it does fail! Removing upload functionality from the edit page.